huaweicloud/services/waf/resource_huaweicloud_waf_policy.go

```diff
@@ -0,0 +1,279 @@
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"basic_web_protection": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"general_check": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_engine": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_scanner": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_script": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"crawler_other": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"webshell": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"cc_attack_protection": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"precise_protection": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"blacklist": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"data_masking": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"false_alarm_masking": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"web_tamper_protection": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"full_detection": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceWafPolicyV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	wafClient, err := config.WafV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud WAF client: %s", err)
+	}
+
+	createOpts := policies.CreateOpts{
+		Name: d.Get("name").(string),
+	}
+	policy, err := policies.Create(wafClient, createOpts).Extract()
+	if err != nil {
+		return fmtp.Errorf("Error creating waf policy: %s", err)
+	}
+
+	logp.Printf("[DEBUG] Waf policy created: %#v", policy)
+	d.SetId(policy.Id)
+	d.Set("name", policy.Name)
+
+	level := d.Get("level").(int)
+	protectionMode := d.Get("protection_mode").(string)
+
+	// Get the policy details, then check if need to update 'protection_mode' or 'level.
+	err = resourceWafPolicyV1Read(d, meta)
+
+	// If the vlaue of 'protection_mode' or 'level' is not equal to the value returned by the server,
+	// then update the policy.
+	err = checkAndUpdateDefaultVal(policy, protectionMode, level, wafClient)
+	if err != nil {
+		return fmtp.Errorf("the Waf Policy was created successfully, "+
+			"but failed to update protection_mode or level : %s", err)
+	} else {
+		d.Set("protection_mode", protectionMode)
+		d.Set("level", level)
+	}
+
+	return err
+}
+
+func checkAndUpdateDefaultVal(policy *policies.Policy, protectionMode string, level int,
+	wafClient *golangsdk.ServiceClient) error {
```

Comment:
suggest to set the golangsdk.ServiceClient as the first parameter

Reply:
done